internal/session/handle_append.go

```diff
@@ -3,32 +3,30 @@ package session
 import (
 	"context"
 	"errors"
-	"time"
 
-	"github.com/ProtonMail/gluon/internal/parser/proto"
+	"github.com/ProtonMail/gluon/imap/command"
 	"github.com/ProtonMail/gluon/internal/response"
 	"github.com/ProtonMail/gluon/internal/state"
 	"github.com/ProtonMail/gluon/profiling"
 	"github.com/ProtonMail/gluon/reporter"
-	"github.com/emersion/go-imap/utf7"
 )
 
-func (s *Session) handleAppend(ctx context.Context, tag string, cmd *proto.Append, ch chan response.Response) error {
+func (s *Session) handleAppend(ctx context.Context, tag string, cmd *command.Append, ch chan response.Response) error {
 	profiling.Start(ctx, profiling.CmdTypeAppend)
 	defer profiling.Stop(ctx, profiling.CmdTypeAppend)
 
-	nameUTF8, err := utf7.Encoding.NewDecoder().String(cmd.GetMailbox())
+	nameUTF8, err := s.decodeMailboxName(cmd.Mailbox)
```

Comment:
Any reason not to auto-decode mailbox names inside the parser? We have a jira ticket for that, it might be nice to do now.

Reply:
Could you store the raw/decoded alongside each other then? Or could you add a method on cmd.Mailbox like `cmd.Mailbox.Decode()`? I really want to try avoid all this decoding code-duplication.

Reply:
I would really like to avoid having to pass the mailbox separator character down the parsing pipeline as it is not really required to complete parsing.

Your suggestion for `cmd.Mailbox.Decode` is no different than what we have at the moment. `decodeMailboxName` does a bit more than just decoding from utf7.